hpack/table.py

```diff
@@ -0,0 +1,215 @@
+    (b'expect'                      , b''             ),
+    (b'expires'                     , b''             ),
+    (b'from'                        , b''             ),
+    (b'host'                        , b''             ),
+    (b'if-match'                    , b''             ),
+    (b'if-modified-since'           , b''             ),
+    (b'if-none-match'               , b''             ),
+    (b'if-range'                    , b''             ),
+    (b'if-unmodified-since'         , b''             ),
+    (b'last-modified'               , b''             ),
+    (b'link'                        , b''             ),
+    (b'location'                    , b''             ),
+    (b'max-forwards'                , b''             ),
+    (b'proxy-authenticate'          , b''             ),
+    (b'proxy-authorization'         , b''             ),
+    (b'range'                       , b''             ),
+    (b'referer'                     , b''             ),
+    (b'refresh'                     , b''             ),
+    (b'retry-after'                 , b''             ),
+    (b'server'                      , b''             ),
+    (b'set-cookie'                  , b''             ),
+    (b'strict-transport-security'   , b''             ),
+    (b'transfer-encoding'           , b''             ),
+    (b'user-agent'                  , b''             ),
+    (b'vary'                        , b''             ),
+    (b'via'                         , b''             ),
+    (b'www-authenticate'            , b''             ))
+
+    def __init__(self):
+        self._maxsize = HeaderTable.DEFAULT_SIZE
+        self.resized = False
+
+    def __getitem__(self, index):
+        """
+        Returns the entry specified by index
+
+        Note that the table is 1-based ie an index of 0 is
+        invalid.  This is due to the fact that a zero value
+        index signals that a completely unindexed header
+        follows.
+
+        The entry will either be from the static table or
+        the dynamic table depending on the value of index.
+        """
+        index -= 1
+        if index < 0:
+            return None # TODO throw HPACKException here
+        if index < len(HeaderTable.STATIC_TABLE):
+            return HeaderTable.STATIC_TABLE[index]
+        index -= len(HeaderTable.STATIC_TABLE)
+        if index < len(self):
+            return deque.__getitem__(self, index)
+        return None # TODO throw HPACKException here
+
+    def __setitem__(self, index, value):
+        """
+        We don't support direct index setting
+        """
+        raise TypeError("'HeaderTable' object does not support item assignment, use add")
+
+    def __repr__(self):
+        rv = "HeaderTable("
+        rv += str(self._maxsize)
+        rv += ", "+str(self.resized)
+        rv += ", ["
+        itm = [str(entry) for entry in self]
+        itm = ", ".join(itm)
+        return rv+itm+"])"
```

Comment:
This would look a bit nicer if it was a big `''.join([])`:

```python
return "".join([
    "HeaderTable(",
    str(self._maxsize),
    ", ",
    str(self.resized),
    ", [",
    ", ".join(str(entry) for entry in self),
    "])"
])
```